api/controllers/loan_controller.js

```diff
@@ -0,0 +1,205 @@
+        
+        const user = users.find(c => c.id === decoded.id);
+        if(!user) return res.status(404).send({"error": "User not found"})
+        if(user.isAdmin !=true) return res.status(401).send({status:401, error: 'You dont have administrative privileges to execute this route.'});
+        const loan = loans.find(c => c.id === parseInt(req.params.id));
+        if(!loan) res.status(404).send({'error':'The loan with the given ID was not found.'});
+        if(!req.body.status) res.status(400).send({'error':'No status provided'});
+        if(req.body.status != 'approved' && req.body.status != 'rejected') res.status(400).send({'error':'the status should either be approved or rejected'});
+
+        //Update loan status
+        loan.status = req.body.status;
+
+        return res.status(200).json({
+            status: 200,
+            data:loans
+        });
+        });
+    }
+
+    // Create a loan repayment record.
+    static payLoan(req, res) {
+        const token = req.headers['x-access-token'];
+        if (!token) return res.status(401).send({ 'error': 'No token provided', 'status': 400 });
+        
+        jwt.verify(token, config.secret, function(err, decoded) {
+            if (err) return res.status(500).send({ status: 500, error: 'Failed to authenticate token.' });
+
+            const user = users.find(c => c.id === decoded.id);
+            const loan = loans.find(c => c.id === parseInt(req.params.id));
+            if(!loan || !user) return res.status(404).send({'error':'User not found or Loan with the provided Id is not found'})
+            if(user.email != loan.user || user.status !='verified' || loan.status !='approved') return res.status(400).send({'error':'User is not verified or Loan is not approved or is rejected or no loan with the provided id is found'});
+            if(!req.body.amount) return res.status(400).send({'error':'No amount provided'});
+
+            const datatime = new Date();
+            const repayment = new Repayment(repayments.length + 1, decoded.id, datatime ,req.params.id,req.body.amount);
+            repayments.push(repayment);
+            loan.balance = loan.balance - req.body.amount;
+            if(loan.balance <= 0) loan.repaid = true;
+
+            return res.status(201).json({
+                status: 201,
+                data:{
+                    id:repayment.id,
+                    loanId : loan.id,
+                    createdOn : repayment.createdOn,
+                    amount : loan.amount,
+                    monthlyPayment: loan.paymentInstallment,
+                    paidAmount: Repayment.amount,
+                    balance : loan.balance
+                }
+          });       
+    });
+
+    }
+
+     //Get loan repayment history
+    static getLoanRepaymentHistory(req, res) {
+        const token = req.headers['x-access-token'];
+            if (!token) return res.status(401).send({ 'error': 'No token provided', 'status': 400 });
```

Comment:
I noticed you have used this line in multiple places in your code. You should consider extracting it to a function that handles token validation then you can reference it wherever you need it.